worldeditadditions/utils/parse/axes_parser.lua

```diff
@@ -0,0 +1,204 @@
+-- ██████   █████  ██████  ███████ ███████      █████  ██   ██ ███████ ███████
+-- ██   ██ ██   ██ ██   ██ ██      ██          ██   ██  ██ ██  ██      ██
+-- ██████  ███████ ██████  ███████ █████       ███████   ███   █████   ███████
+-- ██      ██   ██ ██   ██      ██ ██          ██   ██  ██ ██  ██           ██
+-- ██      ██   ██ ██   ██ ███████ ███████     ██   ██ ██   ██ ███████ ███████
+
+-- Error codes: https://kinsta.com/blog/http-status-codes/
+
+--- FOR TESTING ---
+local function unpack(tbl)
+	if table.unpack then
+		return table.unpack(tbl)
+	else return unpack(tbl) end
+end
+---------------
+
+local Vector3
+if worldeditadditions then
+	local wea = worldeditadditions
+	Vector3 = dofile(wea.modpath.."/utils/vector3.lua")
+else
+	Vector3 = require("worldeditadditions.utils.vector3")
+end
+
+local key_instance
+if worldeditadditions then
+	local wea = worldeditadditions
+	key_instance = dofile(wea.modpath.."/utils/parse/key_instance.lua")
+else
+	key_instance = require("worldeditadditions.key_instance")
+end
+
+--- Unified Axis Keywords banks
+local keywords = {}
+-- Direction keywords
+keywords.dir = {
+	["?"] = "front", f = "front",
+	facing = "front", front = "front",
+	b = "back", back = "back",
+	behind = "back", rear = "back",
+	l = "left", left = "left",
+	r = "right", right = "right",
+	u = "up", up = "up",
+	d = "down", down = "down",
+}
+-- Mirroring keywords
+keywords.mirroring = {
+	sym = true, symmetrical = true,
+	mirror = true, mir = true,
+	rev = true, reverse = true,
+	["true"] = true
+}
+
+--- Initialize parser function container
+local parse = {}
```

Comment:
Given that the parent `init.lua` is `worldeditadditions.parse`, this is confusing. I suggest calling this `axis_parser` instead to avoid confusion.

Reply:
I'm confused. Eventually `worldeditadditions.parse` will be only this and `axis_parser` would be confusing given that this parses numbers, does vectorization and evaluates dirs and other keywords.

Reply:
No, there are other parsers that were implemented before the return paradigm we've since adopted that I need to convert to the new style. See the other files in the `parse` subdirectory. For example:

 - `worldeditadditions.parse.chance()`
 - `worldeditadditions.parse.seed()`
 - `worldeditadditions.parse.map()`
 - `worldeditadditions.parse.tokenise_commands()`

Reply:
Yup, I'm refactoring it.

Reply:
But I really don't want to change the local.